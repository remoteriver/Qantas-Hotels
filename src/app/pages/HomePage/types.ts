import { Color } from '@material-ui/core';
import { Hotel } from 'types/Hotel';

/* --- STATE --- */
export interface HomePageState {
  results: Hotel[];
}

export type ContainerState = HomePageState;

export interface Data {
  results: Hotel[];
}
