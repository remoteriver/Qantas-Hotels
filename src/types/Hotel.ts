export type Hotel = {
  id: string;
  property: Property;
  offer: Offer;
};

export type Property = {
  propertyId: string;
  title: string;
  address: string[];
  previewImage: PreviewImage;
  rating: Rating;
};

export type PreviewImage = {
  url: string;
  caption: string;
  imageType: ImageType;
};

export enum ImageType {
  PRIMARY = 'PRIMARY',
}

export type Rating = {
  ratingValue: number;
  ratingType: RatingType;
};

export enum RatingType {
  SELF = 'self',
  STAR = 'STAR',
}

export type Offer = {
  promotion: Promotion;
  name: string;
  displayPrice: Price;
  savings?: Price;
  cancellationOption: CancellationOption;
};

export type Promotion = {
  title: string;
  type: PromotionType;
};

export enum PromotionType {
  MEMBER = 'MEMBER',
}

export type Price = {
  amount: number;
  currency: Currency;
};

export enum Currency {
  AUD = 'AUD',
}

export type CancellationOption = {
  cancellationType: CancellationType;
};

export enum CancellationType {
  FREE_CANCELLATION = 'FREE_CANCELLATION',
  NOT_REFUNDABLE = 'NOT_REFUNDABLE',
}

/**
 * "id": "cxd650nuyo",
      "property": {
        "propertyId": "P107801",
        "title": "Courtyard by Marriott Sydney-North Ryde",
        "address": [
          "7-11 Talavera Rd",
          "North Ryde"
        ],
        "previewImage": {
          "url": "https://unsplash.it/145/125/?random",
          "caption": "Image of Courtyard by Marriott Sydney-North Ryde",
          "imageType": "PRIMARY"
        },
        "rating": {
          "ratingValue": 4.5,
          "ratingType": "self"
        }
      },
      "offer": {
        "promotion": {
          "title": "Exclusive Deal",
          "type": "MEMBER"
        },
        "name": "Deluxe Balcony Room",
        "displayPrice": {
          "amount": 329.000000000,
          "currency": "AUD"
        },
        "savings": {
          "amount": 30.000000000,
          "currency": "AUD"
        },
        "cancellationOption": {
          "cancellationType": "NOT_REFUNDABLE"
        }
      }
 */
