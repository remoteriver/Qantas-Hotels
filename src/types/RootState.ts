// [IMPORT NEW CONTAINERSTATE ABOVE] < Needed for generating containers seamlessly

import { HomePageState } from 'app/pages/HomePage/types';

/* 
  Because the redux-injectors injects your reducers asynchronously somewhere in your code
  You have to declare them here manually
*/
export interface RootState {
  homepageState?: HomePageState;
  // [INSERT NEW REDUCER KEY ABOVE] < Needed for generating containers seamlessly
}
