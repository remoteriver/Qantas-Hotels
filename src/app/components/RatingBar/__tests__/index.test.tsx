import { render } from '@testing-library/react';
import { RatingType } from 'types/Hotel';
import { RatingBar } from '..';

describe('<HotelCard />', () => {
  const mockRating = {
    ratingValue: 4.5,
    ratingType: RatingType.SELF,
  };

  beforeAll(() => {});

  it('should render and match the snapshot', () => {
    //act
    const component = render(<RatingBar rating={mockRating}></RatingBar>);

    //assert
    expect(component.baseElement).toMatchSnapshot();
  });
});
