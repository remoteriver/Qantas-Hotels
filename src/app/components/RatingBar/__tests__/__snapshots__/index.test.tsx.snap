// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HotelCard /> should render and match the snapshot 1`] = `
<body>
  <div>
    <span
      aria-label="4.5 Stars"
      class="MuiRating-root MuiRating-readOnly"
      role="img"
    >
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="MuiRating-decimal"
      >
        <span
          style="width: 50%; overflow: hidden; z-index: 1; position: absolute;"
        >
          <span
            class="MuiRating-icon MuiRating-iconFilled"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
        <span>
          <span
            class="MuiRating-icon MuiRating-iconEmpty"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
        </span>
      </span>
    </span>
  </div>
</body>
`;
